dojo/product/views.py

```diff
@@ -617,27 +617,47 @@ def view_engagements(request, pid, engagement_type="Interactive"):
 
     default_page_num = 10
 
+    base_engagements = Engagement.objects.filter(
+        product=prod,
+        active=True,
+        engagement_type=engagement_type
+    ).order_by(
+        '-updated'
+    ).select_related(
+        'lead'
+    ).prefetch_related(
+        'tags',
+    ).annotate(
+        count_tests=Count('test', distinct=True),
+        count_findings_all=Count('test__finding__id'),
+        count_findings_open=Count('test__finding__id', filter=Q(test__finding__active=True)),
+        count_findings_open_verified=Count('test__finding__id', filter=Q(test__finding__active=True) & Q(test__finding__verified=True)),
+        count_findings_close=Count('test__finding__id', filter=Q(test__finding__is_Mitigated=True)),
+        count_findings_duplicate=Count('test__finding__id', filter=Q(test__finding__duplicate=True)),
+        count_findings_accepted=Count('test__finding__id', filter=Q(test__finding__risk_accepted=True)),
+    )
+
+    if System_Settings.objects.get().enable_jira:
+        base_engagements = base_engagements.prefetch_related(
+            'jira_project__jira_instance',
+            'product__jira_project_set__jira_instance',
+        )
+
```

Comment:
the idea is to have prefetching in its own method so that we can reuse it elsewhere if needed.

Reply:
I assumed it previously was used in more than one place, but no longer is - but I can certainly put back in a separate function.